src/backend/InvenTree/InvenTree/sso.py

```diff
@@ -75,3 +82,53 @@ def registration_enabled() -> bool:
 def auto_registration_enabled() -> bool:
     """Return True if SSO auto-registration is enabled."""
     return str2bool(get_global_setting('LOGIN_SIGNUP_SSO_AUTO'))
+
+
+def ensure_sso_groups(sender, sociallogin: SocialLogin, **kwargs):
+    """Sync groups from IdP each time a SSO user logs on.
+
+    This event listener is registered in the apps ready method.
+    """
+    if not get_global_setting('LOGIN_ENABLE_SSO_GROUP_SYNC'):
+        return
+
+    group_key = get_global_setting('SSO_GROUP_KEY')
+    group_map = json.loads(get_global_setting('SSO_GROUP_MAP'))
+    # map SSO groups to InvenTree groups
+    group_names = []
+    for sso_group in sociallogin.account.extra_data[group_key]:
+        group_names.append(group_map[sso_group])
+
+    # ensure user has groups
+    user = sociallogin.account.user
+    for group_name in group_names:
+        try:
+            user.groups.get(name=group_name)
+        except Group.DoesNotExist:
+            # user not in group yet
+            try:
+                user.groups.add(Group.objects.get(name=group_name))
+                logger.info(f'Adding group {group_name} to user {user}')
+            except Group.DoesNotExist:
+                logger.error(
+                    f'Failed to add group {group_name} to user {user}: Group does not exist'
+                )
+
+    # remove groups not listed by SSO if not disabled
+    if get_global_setting('SSO_REMOVE_GROUPS'):
+        for group in user.groups.all():
+            if not group.name in group_names:
+                logger.warning(f'Removing group {group.name} from {user}')
```

Comment:
Comment (blocking): In my opinion it would make more sense to log this as an info - this is an intended function and does not require remidiation by an adding. Thoughts @SchrodingersGat?

Reply:
Done :+1: